src/main/java/org/tron/core/services/http/solidity/SolidityNodeHttpApiService.java

```diff
@@ -0,0 +1,110 @@
+import org.tron.core.services.http.GetNowBlockServlet;
+import org.tron.core.services.http.GetPaginatedAssetIssueListServlet;
+import org.tron.core.services.http.ListWitnessesServlet;
+
+@Component
+@Slf4j
+public class SolidityNodeHttpApiService implements Service {
+
+  private int port = Args.getInstance().getSolidityHttpPort();
+
+  private Server server;
+
+  @Autowired
+  private GetAccountServlet accountServlet;
+
+
+  @Autowired
+  private GetTransactionByIdSolidityServlet getTransactionByIdServlet;
+  @Autowired
+  private GetTransactionInfoByIdServlet getTransactionInfoByIdServlet;
+  @Autowired
+  private GetTransactionsFromThisServlet getTransactionsFromThisServlet;
+  @Autowired
+  private GetTransactionsToThisServlet getTransactionsToThisServlet;
+
+  @Autowired
+  private ListWitnessesServlet listWitnessesServlet;
+  @Autowired
+  private GetAssetIssueListServlet getAssetIssueListServlet;
+  @Autowired
+  private GetPaginatedAssetIssueListServlet getPaginatedAssetIssueListServlet;
+  @Autowired
+  private GetNowBlockServlet getNowBlockServlet;
+  @Autowired
+  private GetBlockByNumServlet getBlockByNumServlet;
+
+
+  @Override
+  public void init() {
+
+  }
+
+  @Override
+  public void init(Args args) {
+
+  }
+
+  @Override
+  public void start() {
+    Args args = Args.getInstance();
+    try {
+      server = new Server(port);
+      ServletContextHandler context = new ServletContextHandler(ServletContextHandler.SESSIONS);
+      context.setContextPath("/");
+      server.setHandler(context);
+
+      // same as FullNode
+      context.addServlet(new ServletHolder(accountServlet), "/walletsolidity/getaccount");
+      context.addServlet(new ServletHolder(listWitnessesServlet), "/walletsolidity/listwitnesses");
+      context.addServlet(new ServletHolder(getAssetIssueListServlet),
+          "/walletsolidity/getassetissuelist");
+      context.addServlet(new ServletHolder(getPaginatedAssetIssueListServlet),
+          "/walletsolidity/getpaginatedassetissuelist");
+      context.addServlet(new ServletHolder(getNowBlockServlet), "/walletsolidity/getnowblock");
+      context.addServlet(new ServletHolder(getBlockByNumServlet), "/walletsolidity/getblockbynum");
+
+      // only for SolidityNode
+      context.addServlet(new ServletHolder(getTransactionByIdServlet),
+          "/walletsolidity/gettransactionbyid");
+      context
+          .addServlet(new ServletHolder(getTransactionInfoByIdServlet),
+              "/walletsolidity/gettransactioninfobyid");
+
+      // for extension api
+      if (args.isWalletExtensionApi()) {
+        context.addServlet(new ServletHolder(getTransactionsFromThisServlet),
+            "/walletextension/gettransactionsfromthis");
+        context
+            .addServlet(new ServletHolder(getTransactionsToThisServlet),
+                "/walletextension/gettransactionstothis");
+      }
+
+      server.start();
+    } catch (Exception e) {
+      e.printStackTrace();
```

Comment:
![Vulnerability](https://sonarsource.github.io/sonar-github/vulnerability.png 'Vulnerability') Vulnerability: Use a logger to log this exception. ([squid:S1148](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS1148))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=1114&issues=AWRjYLfoMt9UOhrw8p6f&open=AWRjYLfoMt9UOhrw8p6f)

[ISSUE_KEY]: AWRjYLfoMt9UOhrw8p6f